density_model_tf.py

```diff
@@ -0,0 +1,142 @@
+import argparse
+import tensorflow as tf
+
+import models_tf as models
+import utils
+
+
+def optimistic_restore(session, save_file):
```

Comment:
Is it possible to move this function (optimistic_restore) into utils?

Reply:
optimistic_restore is library specific, so I think it should be kept within the TensorFlow files. We can move it to models though.